src/main/scala/io/iohk/ethereum/network/rlpx/RLPxConnectionHandler.scala

```diff
@@ -136,21 +140,84 @@ class RLPxConnectionHandler(
         context.parent ! MessageReceived(message)
 
       case Failure(ex) =>
-        log.error(ex, "Cannot decode message")
+        log.error(ex, s"Cannot decode message from $peerId")
     }
 
-    def handshaked(messageCodec: MessageCodec): Receive =
+    /**
+      * Handles sending and receiving messages from the Akka TCP connection, while also handling acknowledgement of
+      * messages sent. Messages are only sent when all Ack from previous messages were received.
+      *
+      * @param messageCodec, for encoding the messages sent
+      * @param messagesNotSent, messages not yet sent
+      * @param cancellableAckTimeout, timeout for the message sent for which we are awaiting an acknowledgement (if there is one)
+      * @param nextSeqNumber, sequence number for the next message to be sent
+      */
+    def handshaked(messageCodec: MessageCodec,
+                   messagesNotSent: Queue[MessageSerializable] = Queue.empty,
+                   cancellableAckTimeout: Option[CancellableAckTimeout] = None,
+                   nextSeqNumber: Int = 0): Receive =
```

Comment:
Do you mean that the `seqNumber` parameter from the `sendMessage` function should be renamed `nextSeqNumber`? Or that this should be called `seqNumber` instead of `nextSeqNumber`?